### PR TITLE
Add library code and deps

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/1902

Portforwarding code extracted from helmclient, will be needed in chart-operator e2e test for connecting to the CNR service running in the cluster.

### CODE REVIEW

File: tunnel.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		Namespace: namespace,
+		PodName:   podName,
+		Remote:    remote,
+		stopChan:  make(chan struct{}, 1),
+		readyChan: make(chan struct{}, 1),
+		Out:       ioutil.Discard,
+	}
+}
+
+// close disconnects a tunnel connection.
```
Comment (rossf7) [inline tunnel.go]: Close

File: tunnel.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+func (t *Tunnel) Close() {
+	close(t.stopChan)
+}
+
+// forwardPort opens a tunnel to a kubernetes pod.
```
Comment (rossf7) [inline tunnel.go]: ForwardPort

**Review by xh3b4sd — APPROVED** [review]

**Review by MarcelMue — APPROVED** [review]

**Review by rossf7 — APPROVED** [review]
LGTM after a couple of GoDoc casing issues. Nice that this is extracted so it can be reused.

But also looks like branch protection isn't enabled.  AFAIK I think creating the repo with repo-init can help with this.

https://github.com/giantswarm/repo-init

File: tunnel.go (hunk 1 above)
Comment (fgimenez) [reply tunnel.go]: Sure, done.

File: tunnel.go (hunk 2 above)
Comment (fgimenez) [reply tunnel.go]: Done.

**Comment by fgimenez** [comment]
@rossf7 thx a lot for the pointer to repo-init, looks great will take into account next time, I've manually protected master here.

**Pull request merged**